framework/src/org/checkerframework/common/wholeprograminference/WholeProgramInferenceScenesHelper.java

```diff
@@ -383,11 +384,11 @@ private boolean shouldIgnore(
         ImplicitFor implicitFor = elt.getAnnotation(ImplicitFor.class);
         if (implicitFor != null) {
             TypeKind[] types = implicitFor.types();
+            TypeKind atmKind = atm.getUnderlyingType().getKind();
             for (TypeKind tk : types) {
-                if (tk == atm.getKind()) {
-                    return true;
-                }
+                if (tk == atmKind) return true;
             }
+            // TODO: Handle implicitFor.literals()
```

Comment:
I presume this means that if an annotation is implicit for literals, and whole program inference has inferred it for a literal, then it shouldn't be output into the source code because doing so would only produce clutter.  Is that right?
Does whole program inference ever annotate literals?  I expect it to mostly annotate method signatures rather than method calls.